### ISSUE TITLE
Eliminate UserWarnings for new projects by updating default config.yml 

### ISSUE THREAD

**Issue by TomV** [RasaHQ/rasa#8945]
**Rasa version**: 2.7.1

**Python version**: 3.7.10

**Operating system** (windows, osx, ...): osx Darwin-20.5.0-x86_64-i386-64bit

**Issue**: The default project default configuration result in UserWarning messages (first one, then when that's resolved, a different one.) This is not a good "first time" experience, since as new rasa user, you don't know if the warning is important or if it can be ignored. So you may go spelunking into the docs to figure out what the heck it means, rather than getting basic "hello rasa" example up and running.

The related problem, is that the config.yml file does not actually start with any configuration setting at all – all settings are commented out and the actual values are "inherited" from various places. 'config.yml` settings should be uncommented to use , and the primary source of truth for how the project is configured. Otherwise, the user has to figure out where each configuration setting value has set the default value, etc. There should be one source of truth, and config.yml seems like the obvious place for it. See [Rule of Least Surprise](https://homepage.cs.uri.edu/~thenry/resources/unix_art/ch11s01.html)

I've updated the config.yml file default to address these issues. There is some related documentation that also needs to be updated to reflect that the configuration settings in config.yml are now live.

**Error (including full traceback)**:

(Not Error, just Warning... but WHY have a warning, just fix the default config!) 1\. UserWarning: model_confidence is set to softmax. It is recommended to try using model_confidence=linear_norm to make it easier to tune fallback thresholds. category=UserWarning,

2: /venv/lib/python3.7/site-packages/rasa/utils/train_utils.py:633: UserWarning: constrain_similarities is set to `False`. It is recommended to set it to `True` when using cross-entropy loss. It will be set to `True` by default, Rasa Open Source 3.0.0 onwards.

**Command or request that led to error**:

(again Warning, not error) rasa init rasa train

**Content of configuration file (config.yml)** (if relevant):

<pre>(current default config.yaml after rasa init to create project)</pre>

I'm creating a pull request to address the the issue.

**Comment by sara-tagger** [comment]
Thanks for the issue, @kedz will get back to you about it soon!

###### You may find help in the [docs](https://rasa.com/docs/) and the [forum,](https://forum.rasa.com/) too 🤗

**Comment by TyDunn** [comment]
@dakshvar22 Are these warnings related to the discussion we were having earlier this week?

**Comment by dakshvar22** [comment]
@TyDunn I am not sure which discussion you are referring to 😅 
Warning related `model_confidence` will not come up anymore in Rasa `2.8.0`.
Warning related to `constrain_similarities` shouldn't come up for a new `rasa init` project since auto-configuration already uses `constrain_similarities=True` afaik.

**Comment by TyDunn** [comment]
@dakshvar22 Sorry for the unclear question. We are going to create a test for the auto-configuration that confirms no warnings are created from the default rasa init when training or loading a model, so we will verify that and then close this issue

**Comment by TomV** [comment]
@TyDunn , @dakshvar22  Adding a test to catch warnings for the auto-config sounds like a good idea.  I opened a more specific issue that I think would be closed by doing that: #8975

**Comment by TomV** [comment]
To follow  up on my previous comment, unfortunately, I created a few related issues, because there are different levels to the problem here.   At first glance it would look like these are redundant issues, but one is tactical, and one is a higher level issue.    I think I should have created just two issues, but I created three 🙀   ... 

These are the three overlapping, but somewhat separate issues:
* #8945 (this issue) "Eliminate UserWarnings for new projects by updating default config.yml"
* #8975 "Multiple Unnecessary UserWarnings on new project created with rasa init"
* #8976 "Use an active config.yml file (not commented out) on new project (rasa init) "
                 There is a related pull request that includes valid, uncommented config.yml for the default project file that does not emit any UserWarnings.  However, given the complexity of how the "suggested" config values are dynamically written, it seems this is an incomplete solution (more clean up of the previous approach needs to be done.) (See my comment about this here: https://github.com/RasaHQ/rasa/issues/8976#issuecomment-870894536 )

Sorry I have made a bit of a mess out of the issue.  I started with one thing and realized it is actually a few different things.  Trying to make the user/developer experience smoother and more intuitive..    Hope this summary helps.  

One challenge is that the auto assignment of the github issues might mean that my overlapping issues are not seen as a whole.  Ideally, someone with a better big picture of the code, and architecture of Rasa could probably clean up these into a simpler and clearer set of issues, that would address my concerns.  (one reason for this comment that exposes and consolidates my redundant attempts to address the topic.)
